### ISSUE TITLE
Being able to follow individual accounts from blocked instances

### ISSUE THREAD

**Issue by Mbodin** [mastodon/mastodon#21798]
### Pitch

If a user blocked an instance, he/she should still be able to follow individuals from this instance.
Currently, this user gets a 403 error without any explanation of why following this account is not allowed.

This might be related to #9770, but it seems to me that this is a different issue.

### Motivation

I have blocked some instances (too big with respect to their moderation team), and I noticed that this really helps with my mental health. I still follow some people from these instances, though: Mastodon seems to cope with this relatively fine.

However, if I want to follow an additional person from such an instance (which frequently happens as these are big instances), I get a 403 error. This is confusing.

It is confusing because I don’t get any explanation of why this operation isn’t allowed (I would like to at least have a message stating that it can’t happen because I blocked the instance). It is also confusing because I still follow people from this instance: the sequence of “unblocking the instance, following this additional account, then re-blocking the instance” is permitted, and seems exactly what I’m looking for (although very cumbersome as-is). I would thus expect the interface to enable such a follow.

**Comment by trwnh** [comment]
#14959

**Comment by Mbodin** [comment]
I agree that these feature requests come with a similar aspect, but to me they are different:
* The first one was aimed for instances with non-offensive content (just something the user is not interested in), and only applies on the public timeline. In my case, I block to specifically hide responses from such an instance. But because the instance is large, not everything is to be hidden, and I’m basically trying to build a whitelist.
* The second would indeed technically solve the issue, but with a technical flavour: I don’t feel that filters is easy to use for most people, just by the fact that it’s in the preferences menu. I feel that in this case, this should be an item next to the “Block instance” button.

Besides, this does not change the fact that the 403 error displayed when trying to follow someone is confusing in this situation: I marked it as a suggestion because I understands why, but it took me some time to understand, and I think that a new user would typically mark it as a bug.
